ui/desktop/electron-app/src/services/client-agent-daemon-manager.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+const { spawnSync, spawn } = require('../helpers/spawn-promise');
+
+// TODO: Rename commands from ferry to client-agent
+
+class ClientAgentDaemonManager {
+  /**
+   * Checks the status of the client agent daemon.
+   * @returns {string}
+   */
+  status() {
+    const clientAgentStatusCommand = ['ferry', 'status', '-format=json'];
+    const { stdout } = spawnSync(clientAgentStatusCommand);
+
+    return JSON.parse(stdout);
```

Comment:
Note that I started using `JSON.parse` rather than `jsonify` here. `jsonify` was swallowing any errors that occurred during parsing which I was not a fan of. I'd rather we bubble up the error and we can let the client choose what to do with the error (most likely just log it).

Reply:
I went back to jsonify and decided I'll log the error when we add logging since I need to parse both stdout and stderr if stdout doesn't have parse-able output